### PR TITLE
[Lang] Add math module to support glsl-style functions

### PR DESCRIPTION
**Pull request by zemora**

### CODE REVIEW

**Review by k-ye — APPROVED** [review]
Fantastic LGTM!

File: python/taichi/math/mathimpl.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
+"""
+Math functions for glsl-like functions and other stuff.
+"""
+from math import e, pi
+
+import taichi as ti
+
+
+@ti.func
+def mix(x, y, a):
```
Comment (YuCrazing) [inline python/taichi/math/mathimpl.py:10]:
```suggestion
def lerp(x, y, a):
```

Comment (YuCrazing) [reply python/taichi/math/mathimpl.py:10]: I believe that `lerp` is a more common name. :)

**Review by YuCrazing — CHANGES_REQUESTED** [review]

File: python/taichi/math/mathimpl.py (hunk 1 above)
Comment (neozhaoliang) [reply python/taichi/math/mathimpl.py:10]: I guess `lerp` is used mostly for HLSL users? It's not in the GLSL spec: 
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/

**Pull request merged**